mmdet/models/task_modules/samplers/multi_instance_random_sampler.py

```diff
@@ -0,0 +1,130 @@
+        replaced by `MultiInsRandomSampler`
+    """
+
+    def _sample_pos(self, assign_result: AssignResult, num_expected: int,
+                    **kwargs) -> Union[Tensor, ndarray]:
+        """Randomly sample some positive samples.
+
+        Args:
+            assign_result (:obj:`AssignResult`): Bbox assigning results.
+            num_expected (int): The number of expected positive samples
+
+        Returns:
+            Tensor or ndarray: sampled indices.
+        """
+        pos_inds = torch.nonzero(
+            assign_result.labels[:, 0] > 0, as_tuple=False)
+        if pos_inds.numel() != 0:
+            pos_inds = pos_inds.squeeze(1)
+        if pos_inds.numel() <= num_expected:
+            return pos_inds
+        else:
+            return self.random_choice(pos_inds, num_expected)
+
+    def _sample_neg(self, assign_result: AssignResult, num_expected: int,
+                    **kwargs) -> Union[Tensor, ndarray]:
+        """Randomly sample some negative samples.
+
+        Args:
+            assign_result (:obj:`AssignResult`): Bbox assigning results.
+            num_expected (int): The number of expected positive samples
+
+        Returns:
+            Tensor or ndarray: sampled indices.
+        """
+        neg_inds = torch.nonzero(
+            assign_result.labels[:, 0] == 0, as_tuple=False)
+        if neg_inds.numel() != 0:
+            neg_inds = neg_inds.squeeze(1)
+        if len(neg_inds) <= num_expected:
+            return neg_inds
+        else:
+            return self.random_choice(neg_inds, num_expected)
+
+    def sample(self, assign_result: AssignResult, pred_instances: InstanceData,
+               gt_instances: InstanceData,
+               **kwargs) -> MultiInstanceSamplingResult:
+        """Sample positive and negative bboxes.
+
+        Args:
+            assign_result (:obj:`AssignResult`): Assigning results from
+                MultiInstanceAssigner.
+            pred_instances (:obj:`InstanceData`): Instances of model
+                predictions. It includes ``priors``, and the priors can
+                be anchors or points, or the bboxes predicted by the
+                previous stage, has shape (n, 4). The bboxes predicted by
+                the current model or stage will be named ``bboxes``,
+                ``labels``, and ``scores``, the same as the ``InstanceData``
+                in other places.
+            gt_instances (:obj:`InstanceData`): Ground truth of instance
+                annotations. It usually includes ``bboxes``, with shape (k, 4),
+                and ``labels``, with shape (k, ).
+
+        Returns:
+            :obj:`MultiInstanceSamplingResult`: Sampling result.
+        """
+
+        assert 'batch_gt_instances_ignore' in kwargs, \
+            'batch_gt_instances_ignore is necessary for MultiInsRandomSampler'
+
+        gt_bboxes = gt_instances.bboxes
+        ignore_bboxes = kwargs['batch_gt_instances_ignore'].bboxes
```

Comment:
If we need batch_gt_instances_ignore here, why not add set as default and do not use kwargs?

Reply:
This place can really simplify, but I want to make the formal parameters as identical as possible to the parameters of the parent class(`BaseSampler`). If simplify the code here, the name of the parameter may change.